### PR TITLE
Password check

### PR DESCRIPTION
**Pull request by Rogal255**
Password checker with bonus excercise

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @Rogal255

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @Rogal255

**Comment by ziobron** [comment]
⏱ 3 XP extra for @Rogal255 for on-time delivery

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,74 @@
...
+    }
+    else if (std::none_of(t_Pass.cbegin(), t_Pass.cend(), [](const unsigned char c){ return std::isdigit(c); }))
+    {
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
+    }
+    else if(std::none_of(t_Pass.cbegin(), t_Pass.cend(), [](const unsigned char c){ return std::ispunct(c); }))
+    {
+        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+    }
+    else if(std::none_of(t_Pass.cbegin(), t_Pass.cend(), [](const unsigned char c){ return std::isupper(c); }))
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:60]: When you use return inside `if` you don't need `if else` or `else`. Just write `if (sth) { return blabla; }`

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,74 @@
...
+    {
+        return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+    }
+
+    return ErrorCode::Ok;
+}
+
+ErrorCode checkPassword(const std::string& t_FirstPass, const std::string& t_SecondPass){
+    if(!doPasswordsMatch(t_FirstPass, t_SecondPass))
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:69]: Add space after `if`, `for`, `while` `function` etc...

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**